app/src/createExcelFiles/fetchSearches.js

```diff
@@ -0,0 +1,25 @@
+import { SearchItem } from "./modal";
+import { mapToExcelArray } from "./ExcelUtils";
+
+export async function fetchSearches(searchName, searchNumber) {
+  const response = await (
+    await fetch(
+      `https://google-search3.p.rapidapi.com/api/v1/search/q=${searchName}&num=${searchNumber}`,
+      {
+        method: "GET",
+        headers: {
+          "x-rapidapi-key": process.env.REACT_APP_API_KEY,
+          "x-proxy-location": "US",
+        },
+      }
+    )
+  ).json();
+
+  const searchItems = await response.results.map(
+    (item) => new SearchItem(item.title, item.description, item.link)
+  );
+  
+  const arrayOfArray = mapToExcelArray(searchItems);  
```

Comment:
generally, this is not a good design.

This is your fetchSearches method, which should return in the *model* (SearchItem) since that is what is understood by the app.

This can be taken by your Box.js and THEN passed to createExcel utils, which can use the `mapToExcelArray` locally in the method.

The **model** is what ur supposed to use to communicate between different classes of your app.

So basically,

Box.js -> FetchSearches
FetchSearches --(model)--> Box.js
Box.js --(model)--> CreateExcel
CreateExcel --(model)--> mapToExcelArray
mapToExcelArray --(excel specific model)--> CreateExcel 


THAT is considered the right design pattern